c/driver/postgresql/connection.cc

```diff
@@ -68,16 +69,54 @@ class PqResultRow {
 
 // Helper to manager the lifecycle of a PQResult. The query argument
 // will be evaluated as part of the constructor, with the desctructor handling cleanup
-// Caller is responsible for calling the `Status()` method to ensure results are
-// as expected prior to iterating
+// Caller must call Prepare then Execute, checking both for an OK AdbcStatusCode
+// prior to iterating
 class PqResultHelper {
  public:
-  PqResultHelper(PGconn* conn, const char* query) : conn_(conn) {
+  explicit PqResultHelper(PGconn* conn, const char* query, struct AdbcError* error)
+      : conn_(conn), error_(error) {
     query_ = std::string(query);
-    result_ = PQexec(conn_, query_.c_str());
+    param_values_ = {};
   }
 
-  ExecStatusType Status() { return PQresultStatus(result_); }
+  explicit PqResultHelper(PGconn* conn, const char* query,
```

Comment:
I guess, why not just declare it as taking a `std::string` and then initializing `query_` with a move?